### DISCUSSION TITLE
Display Image from File Upload

### DISCUSSION THREAD

**Discussion by ecxs** [Budibase/budibase#9160] [Show and tell]
In https://github.com/Budibase/budibase/discussions/9136 @melohagan referred me to the BudiBase File Upload plugin.

Inspired by the concept of storing a file as base64 in a json object inside a text field, I then started thinking about how it would be possible to display the stored file.

I then started tinkering around, and I finally found a simple solution. With the help of an embed component and a little JavaScript, I managed to pull this off. I'm not fully up to speed with JS yet, and I'm not sure whether this is anything useful, but I just wanted to share this with my fellow BB developers ... just in case. And if it's not of any use to you, then it's a least a little showcase of how flexible and powerful BB is.

This is the JavaScript:

```js
var base = $("New Repeater.TestBB.FileText");
var base = JSON.parse(base.slice(2,base.length -2));
var base = base.link;
var base = base.slice(base.indexOf(',')+1);
var type = base.type;
var div = '<div><img src="data:' + type + ';base64, ' + base + '"/> </div>';
return div;
```

And the result being:

![Screenshot 2022-12-23 at 08 41 32](https://user-images.githubusercontent.com/4005405/209294740-777ce4f7-4087-44af-8f78-3b5c90802dbe.png)

**Comment by acseven** [comment]
This is great!
